features/steps/thens/geometry.py

```diff
@@ -13,16 +13,8 @@ def step_impl(context, inst, clause):
     entity_contexts = ifc.recurrently_get_entity_attr(context, inst, 'IfcRepresentation', 'ContextOfItems')
     precision = ifc.get_precision_from_contexts(entity_contexts)
     points_coordinates = geometry.get_points(inst)
-    comparison_nr = 1
-    duplicates = set()
-    for i in itertools.combinations(points_coordinates, 2):
-        if math.dist(i[0], i[1]) < precision:
-            if clause == 'including' or (clause == 'excluding' and comparison_nr != len(points_coordinates) - 1):
-                # combinations() produces tuples in a sorted order, first and last item is compared with items 0 and n-1
-                duplicates.add(i)
-                if len(duplicates) > 2:  # limit nr of reported duplicate points to 3 for error readability
-                    break
-        comparison_nr += 1
-    if duplicates:
-        yield ValidationOutcome(inst=inst, expected="No duplicates", observed="Duplicates", severity=OutcomeSeverity.ERROR)
-
+    for i, j in itertools.combinations(range(len(points_coordinates)), 2):
+        # combinations() produces tuples in a sorted order, first and last item is compared with items 0 and n-1
+        if clause == 'including' or (clause == 'excluding' and (i, j) == (0, len(points_coordinates) - 1)):
```

Comment:
```suggestion
        if clause == 'including' or (clause == 'excluding' and (i, j) =! (0, len(points_coordinates) - 1)):
```
Shouldn't this be `!= `instead?
Changing this causes the tests to pass. If the cause is excluding and the point is equal to the last point, we'd like to return False so there is no comparison.